src/rust/collections/pin_slab.rs

```diff
@@ -107,22 +114,23 @@ impl<T> PinSlab<T> {
         self.internal_get(key)
     }
 
-    /// Insert a value into the pin slab.
-    pub fn insert(&mut self, val: T) -> Option<usize> {
-        let key: usize = self.next;
-        self.insert_at(key, val)?;
-        Some(key)
+    #[allow(unused)]
+    /// Gets a read only pinned reference.
+    pub fn get_pin(&self, key: usize) -> Option<Pin<&T>> {
+        let entry: &T = self.internal_get(key)?;
+        // Safety: all storage is pre-allocated in chunks, and each chunk
+        // doesn't move. We only provide mutators to drop the storage through
+        // `remove` (but it doesn't return it).
+        unsafe { Some(Pin::new_unchecked(entry)) }
```

Comment:
If unused, we can get rid of this.

Reply:
We may want to use this in some other code, I left it for symmetry since we have get_pin_mut and get (and I didn't want to re-write this function later if we need it).